### PR TITLE
[skip ci] Use @tenstorrent/metalium-developers-infra in CODEOWNERS

### PR DESCRIPTION
**Pull request by blozano-tt**
### Problem description
@tt-rkim was frequently the only code owner on several PRs. This means he can never take vacation.

### What's changed
Use the infra team as codeowners for shared ownership of PR review.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -181,11 +171,10 @@ tests/device_perf_tests/stable_diffusion/test_perf_stable_diffusion.py @esmalTT
 tests/ttnn/integration_tests/unet @esmalTT @uaydonat @mywoodstock
 tests/nightly/wh_b0_only_eth/experimental/functional_unet @esmalTT @uaydonat @mywoodstock
 scripts/profiler/ @mo-tenstorrent
-scripts/docker @ttmchiou @tt-rkim
+scripts/docker @tenstorrent/metalium-developers-infra
 
-dockerfile @ttmchiou @tt-rkim
+dockerfile @tenstorrent/metalium-developers-infra
 
-tt_metal/CMakeLists.txt @abhullar-tt @pgkeller @aliuTT @tt-aho @tt-dma @tt-asaigal @ubcheema @blozano-tt
```
Comment (blozano-tt) [inline CODEOWNERS:188]: I don't want to own this so much.

File: CODEOWNERS [hunk 2]
```diff
@@ -164,12 +157,9 @@ models/perf/perf_report.py @yieldthought @uaydonat @tt-rkim
 models/perf/benchmarking_utils.py @skhorasganiTT @tt-rkim
 
 # docs
+docs/Makefile @tenstorrent/metalium-developers-infra
 docs/source/ttnn/dependencies/tt_lib.rst @eyonland @patrickroberts @yan-zaretskiy @ayerofieiev-tt
 docs/source/ttnn/ @eyonland @patrickroberts @yan-zaretskiy @ayerofieiev-tt @razorback3 @dongjin-na
-# docs/source/apis/host_apis/ @abhullar-tt @TT-billteng @davorchap @tt-rkim
-# docs/source/apis/host_apis2.rst @abhullar-tt @TT-billteng @davorchap @tt-rkim
-# docs/source/apis/kernel_apis/ @davorchap @pgkeller @tt-rkim
-# docs/source/apis/kernel_apis.rst @davorchap @pgkeller @tt-rkim
```
Comment (blozano-tt) [inline CODEOWNERS:172]: lets remove dead code

File: CODEOWNERS [hunk 3]
```diff
@@ -164,12 +157,9 @@ models/perf/perf_report.py @yieldthought @uaydonat @tt-rkim
 models/perf/benchmarking_utils.py @skhorasganiTT @tt-rkim
 
 # docs
+docs/Makefile @tenstorrent/metalium-developers-infra
```
Comment (blozano-tt) [inline CODEOWNERS:160]: @tt-rkim was the sole owner before

File: CODEOWNERS (hunk 1 above)
Comment (blozano-tt) [reply CODEOWNERS:188]: Its owned by tt-metal runtime team. By removing this line, it just removes me.

**Review by tt-rkim — APPROVED** [review]

**Comment by blozano-tt** [comment]
> Hi Wilder, based on conversations with Raymond, could you also please swap or remove the other 12 occurrences of tt-rkim as follows: Any time it's Raymond and Utku only, please remove Raymond (he was only there to ensure PRs followed standard post-commit process):
> 
> * Lines 130,134,136,152,155,
> 
> All other lines, please replace tt-rkim with tenstorrent/metalium-developers-infra, as you've done elsewhere.
> 
> * Lines 38,42,79,86,91,156,157
>   Thanks!
> 
> -Bryan :)

Done!

**Pull request merged**